File: Java/Aplikasi-Search-Github/app/src/main/java/com/makhalibagas/searchgithub/Retrofit/ApiService.java
package com.makhalibagas.searchgithub.Retrofit;

import com.makhalibagas.searchgithub.Model.ResponseSearch;
import com.makhalibagas.searchgithub.Model.User;
import com.makhalibagas.searchgithub.Model.Useritem;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {


    @GET("search/users")
    Call<ResponseSearch> getGithubSearch(
            @Query("q") String username
    );


    @GET("users/{username}")
    Call<User> getUserDetail(@Path("username") String username);


    @GET("users/{username}/followers")
    Call<List<Useritem>> getFollowerData(
            @Header("Authorization") String authorization,
            @Path("username") String username
    );

    @GET("users/{username}/following")
    Call<List<Useritem>> getFollowingData(
            @Header("Authorization") String authorization,
            @Path("username") String username
    );

}

File: Kotlin/Submission2/app/src/main/java/com/makhalibagas/submission2/retrofit/ApiService.kt
package com.makhalibagas.submission2.retrofit
import com.makhalibagas.submission2.model.detailUser.DetailUser
import com.makhalibagas.submission2.model.repository.Repository
import com.makhalibagas.submission2.model.user.ResponseUser
import com.makhalibagas.submission2.model.user.User
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by Bagas Makhali on 7/2/2020.
 */
interface ApiService {

    @GET("search/users")
    @Headers("Authorization:b082949e2b8f06ec054563d89049516278fa321c")
    fun getSearchData(
        @Query("q") query: String
    ) : Call<ResponseUser>


    @GET("users/{username}")
    @Headers("Authorization:b082949e2b8f06ec054563d89049516278fa321c")
    fun getDetailUser(
        @Path("username") username : String
    ) : Call<DetailUser>


    @GET("users/{username}/followers")
    @Headers("Authorization:b082949e2b8f06ec054563d89049516278fa321c")
    fun getFollowersData(
        @Path("username") username : String,
        @Query("page") page : String
    ) : Call<List<User>>


    @GET("users/{username}/following")
    @Headers("Authorization:b082949e2b8f06ec054563d89049516278fa321c")
    fun getFollowingData(
        @Path("username") username : String,
        @Query("page") page : String
    ) : Call<List<User>>

    //https://api.github.com/users/sidiqpermana/followers?page=5
    @GET("users/{username}/repos")
    @Headers("Authorization:b082949e2b8f06ec054563d89049516278fa321c")
    fun getRepositoryData(
        @Path("username") username : String
    ) : Call<List<Repository>>

}